step.go

```diff
@@ -0,0 +1,487 @@
+	CertificatePassphraseList stepconf.Secret `env:"passphrase_list"`
+	KeychainPath              string          `env:"keychain_path"`
+	KeychainPassword          stepconf.Secret `env:"keychain_password"`
+	BuildURL                  string          `env:"BITRISE_BUILD_URL"`
+	BuildAPIToken             stepconf.Secret `env:"BITRISE_BUILD_API_TOKEN"`
+	// Step output configuration
+	OutputDir string `env:"output_dir,required"`
+	// Caching
+	CacheLevel string `env:"cache_level,opt[none,swift_packages]"`
+	// Debugging
+	VerboseLog bool `env:"verbose_log,opt[yes,no]"`
+}
+
+type Config struct {
+	ProjectPath            string
+	Scheme                 string
+	Configuration          string
+	Destination            string
+	XCConfigContent        string
+	XcodebuildOptions      []string
+	XCPretty               bool
+	CodesignManager        *codesign.Manager
+	OutputDir              string
+	XcodebuildMajorVersion int
+	CacheLevel             string
+	SwiftPackagesPath      string
+}
+
+type TestBuilder struct {
+}
+
+func NewTestBuilder() TestBuilder {
+	return TestBuilder{}
+}
+
+func (b TestBuilder) ProcessConfig() (Config, error) {
+	var input Input
+	parser := stepconf.NewInputParser(env.NewRepository())
+	if err := parser.Parse(&input); err != nil {
+		return Config{}, err
+	}
+	logger := v2log.NewLogger()
+	logger.EnableDebugLog(input.VerboseLog)
+	log.SetEnableDebugLog(input.VerboseLog)
+
+	stepconf.Print(input)
+	fmt.Println()
+
+	absProjectPath, err := filepath.Abs(input.ProjectPath)
+	if err != nil {
+		return Config{}, fmt.Errorf("failed to expand project path (%s): %w", input.ProjectPath, err)
+	}
+
+	absOutputDir, err := pathutil.AbsPath(input.OutputDir)
+	if err != nil {
+		return Config{}, fmt.Errorf("failed to expand output dir (%s): %w", input.OutputDir, err)
+	}
+
+	if exist, err := pathutil.IsPathExists(absOutputDir); err != nil {
+		return Config{}, fmt.Errorf("failed to check if output dir exist: %w", err)
+	} else if !exist {
+		if err := os.MkdirAll(absOutputDir, 0777); err != nil {
+			return Config{}, fmt.Errorf("failed to create output dir (%s): %w", absOutputDir, err)
+		}
+	}
+
+	factory := v2command.NewFactory(env.NewRepository())
+	xcodebuildVersion, err := utility.GetXcodeVersion()
+	if err != nil {
+		return Config{}, fmt.Errorf("failed to get xcode version: %w", err)
+	}
+	log.Infof("Xcode version: %s (%s)", xcodebuildVersion.Version, xcodebuildVersion.BuildVersion)
+
+	var swiftPackagesPath string
+	if xcodebuildVersion.MajorVersion >= 11 {
+		var err error
+		if swiftPackagesPath, err = cache.SwiftPackagesPath(absProjectPath); err != nil {
+			return Config{}, fmt.Errorf("failed to get swift packages path: %w", err)
+		}
+	}
+
+	var customOptions []string
+	if input.XcodebuildOptions != "" {
+		customOptions, err = shellquote.Split(input.XcodebuildOptions)
+		if err != nil {
+			return Config{}, fmt.Errorf("failed to parse additional options (%s): %w", input.XcodebuildOptions, err)
```

Comment:
Used something like "provided additional options are not valid CLI argumensts" in the npm/yarn Steps on Oliver's reccomendation.